lib/src/excel.dart

```diff
@@ -1,36 +1,128 @@
+  static final _fontName = 'Roboto';
+  static final _fontSize = 12;
+  static final _horizontalAlignment = 'Center';
+  static final _borderStyle = 'Continuous';
 
-  static final ExcelDart _singleton = ExcelDart._private();
+  static final _tableHeader = [
+    'Insertion Order ID',
+    'Advertiser ID',
+    'Campaign ID',
+    'Display Name',
+    'Entity Status',
+    'Update Time',
+    'Pacing Period',
+    'Pacing Type',
+    'Daily Max Micros',
+    'Daily Max Impressions',
+    'Budget Unit',
+    'Automation Type',
+    'Budget',
+    'Start Date',
+    'End Date'
+  ];
 
-  factory ExcelDart() {
-    return _singleton;
-  }
-
-  /// Waits for Office APIs to be ready and then executes function [_populate].
-  void exec() async {
+  /// Waits for Office APIs to be ready and then populates the spreadsheet with
+  /// entries in the [insertionOrderList]
+  static void populate(List insertionOrderList) async {
     await Office.onReady(allowInterop((info) async {
-      await ExcelJS.run(allowInterop(_populate));
+      await ExcelJS.run(allowInterop((context) {
+        // Grabs the current worksheet
+        final sheet = context.workbook.worksheets.getActiveWorksheet();
+
+        // Adds a master table
+        final endAddressInt = _startAddressInt + _tableHeader.length - 1;
+        final tableAddress = 'A1:${String.fromCharCode(endAddressInt)}1';
+        final table = context.workbook.tables.add(tableAddress, true);
+
+        // Adds and formats table header
+        table.getHeaderRowRange()
+          ..values = [_tableHeader]
+          ..format.font.name = _fontName
+          ..format.font.size = _fontSize
+          ..format.font.bold = true
+          ..format.horizontalAlignment = _horizontalAlignment
+          ..format.borders.getItem('EdgeTop').style = _borderStyle
+          ..format.borders.getItem('EdgeBottom').style = _borderStyle;
+
+        // Adds and formats table body
+        table.getDataBodyRange()
+          ..formulas = _generateTableBody(insertionOrderList)
+          ..format.font.name = _fontName
+          ..format.font.size = _fontSize
+          ..format.horizontalAlignment = _horizontalAlignment;
+
+        // Auto-fits all used cells
+        sheet.getUsedRange().getEntireColumn().format.autofitColumns();
+        sheet.getUsedRange().getEntireRow().format.autofitRows();
+
+        // Sets the sheet as active
+        sheet.activate();
+        return context.sync();
+      }));
     }));
   }
 
-  Future<void> _populate(RequestContext context) async {
-    final sheet = context.workbook.worksheets.getActiveWorksheet();
-    final range = sheet.getRange('A1');
-    final values = <List<String>>[
-      ['test test']
-    ];
-    range.values = values;
-    range.format.autofitColumns();
-    return context.sync();
+  static List _generateTableBody(List insertionOrderList) {
+    final tableBody = [];
+
+    for (InsertionOrder io in insertionOrderList) {
```

Comment:
what about doing something like insertionOrderList.map(conversionfunction).toList() do you think that would be more idiomatic?